### PR TITLE
Add scroll interface tabs with mouse wheel

### LINKED ISSUE #282: Unable to scroll interface tabs with mouse wheel on MPM
#### Expected behaviour
On Ubuntu MATE 16.04 LTS user was able to use mouse wheel to scroll interface tabs.

#### Actual behaviour
On development version of Ubuntu 19.10 this is not possible - tabs do not react on mouse wheel.

#### Steps to reproduce the behaviour
1. Install Ubuntu MATE 19.10
2. Open any of `mate-power-preferences`, `mate-power-statistics`
3. Try to switch interface tabs with mouse wheel

#### MATE general version
1.22.1

#### Package version
1.22.1-0ubuntu1

#### Linux Distribution
Ubuntu 19.10

#### Link to downstream report of your Distribution
https://bugs.launchpad.net/ubuntu/+source/mate-power-manager/+bug/1828802

### PR DESCRIPTION
**Pull request by tamplan**
Fixes #282 

mate-power-statistics untested because i have a pc never on battery.

### CODE REVIEW

File: src/gpm-prefs-core.c [hunk 1]
```diff
@@ -607,8 +608,14 @@ prefs_setup_ups (GpmPrefs *prefs)
 					GPM_SETTINGS_SLEEP_DISPLAY_UPS,
 					display_times);
 
+        window = gpm_window (prefs);
+        notebook = GTK_NOTEBOOK (gtk_builder_get_object (prefs->priv->builder, "notebook_preferences"));
+	gtk_widget_add_events (notebook, GDK_SCROLL_MASK);
+        g_signal_connect (notebook, "scroll-event",
+                                        G_CALLBACK (gpm_dialog_page_scroll_event_cb),
```
Comment (raveit65) [inline src/gpm-prefs-core.c]: Indents...

**Comment by raveit65** [comment]
Btw. Maybe you are interested in this repo with MATE 1.23.x RPMs for fedora?
https://copr.fedorainfracloud.org/coprs/raveit65/MATE-1.23.x/

File: src/gpm-prefs-core.c [hunk 2]
```diff
@@ -607,8 +608,14 @@ prefs_setup_ups (GpmPrefs *prefs)
 					GPM_SETTINGS_SLEEP_DISPLAY_UPS,
 					display_times);
 
+        window = gpm_window (prefs);
+        notebook = GTK_NOTEBOOK (gtk_builder_get_object (prefs->priv->builder, "notebook_preferences"));
+	gtk_widget_add_events (notebook, GDK_SCROLL_MASK);
```
Comment (raveit65) [inline src/gpm-prefs-core.c]:
```
gpm-prefs-core.c: In function 'prefs_setup_ups':
gpm-prefs-core.c:613:25: warning: passing argument 1 of 'gtk_widget_add_events' from incompatible pointer type [-Wincompatible-pointer-types]
  613 |  gtk_widget_add_events (notebook, GDK_SCROLL_MASK);
      |                         ^~~~~~~~
      |                         |
      |                         GtkNotebook * {aka struct _GtkNotebook *}
In file included from /usr/include/gtk-3.0/gtk/gtkapplication.h:27,
                 from /usr/include/gtk-3.0/gtk/gtkwindow.h:33,
                 from /usr/include/gtk-3.0/gtk/gtkdialog.h:32,
                 from /usr/include/gtk-3.0/gtk/gtkaboutdialog.h:30,
                 from /usr/include/gtk-3.0/gtk/gtk.h:31,
                 from gpm-prefs-core.c:29:
/usr/include/gtk-3.0/gtk/gtkwidget.h:972:65: note: expected 'GtkWidget *' {aka 'struct _GtkWidget *'} but argument is of type 'GtkNotebook *' {aka 'struct _GtkNotebook *'}
  972 | void       gtk_widget_add_events          (GtkWidget           *widget,
      |                                            ~~~~~~~~~~~~~~~~~~~~~^~~~~~
```

Comment (tamplan) [reply src/gpm-prefs-core.c]: Normally done.

**Comment by tamplan** [comment]
> Btw. maybe you are interested in this repo with MATE 1.23.x RPMs for fedora?
> https://copr.fedorainfracloud.org/coprs/raveit65/MATE-1.23.x/

Back to Ubuntu 19.04, i have tried fedora 30 for glade with success but i prefer synaptic i found more practical than the dnf gui tool.

File: src/gpm-common.h [hunk 3]
```diff
@@ -160,6 +161,9 @@ guint        gpm_discrete_from_percent          (guint       percentage,
 guint        gpm_discrete_to_percent            (guint       discrete,
                                 guint        levels);
 void 		 gpm_help_display				(const gchar	*link_id);
+gboolean        gpm_dialog_page_scroll_event_cb                 (GtkWidget      *widget,
+                                                                 GdkEventScroll *event,
```
Comment (raveit65) [inline src/gpm-common.h]: Indents!

File: src/gpm-prefs-core.c [hunk 4]
```diff
@@ -575,7 +575,8 @@ static void
 prefs_setup_ups (GpmPrefs *prefs)
 {
 	GtkWidget *widget;
-	GtkNotebook *notebook;
+	GtkWidget *notebook;
```
Comment (raveit65) [inline src/gpm-prefs-core.c]: Mixed usage of tabs and spaces. Code use tabs before and after.

File: src/gpm-statistics.c [hunk 5]
```diff
@@ -1681,6 +1681,12 @@ main (int argc, char *argv[])
 			  G_CALLBACK (gpm_stats_points_checkbox_stats_cb), NULL);
 
 	widget = GTK_WIDGET (gtk_builder_get_object (builder, "notebook1"));
+
+	gtk_widget_add_events (widget, GDK_SCROLL_MASK);
```
Comment (raveit65) [inline src/gpm-statistics.c]: Mixed usage of tabs and spaces.

**Comment by raveit65** [comment]
You see that code-style issues easily in browser!
https://patch-diff.githubusercontent.com/raw/mate-desktop/mate-power-manager/pull/291.diff

**Comment by raveit65** [comment]
Beside from that, scrolling with the mouse on notebook tabs works fine in preferences and gpm-statistics (mouse-battery).

File: src/gpm-statistics.c (hunk 5 above)
Comment (tamplan) [reply src/gpm-statistics.c]: Done!

File: src/gpm-statistics.c [hunk 6]
```diff
@@ -1681,6 +1681,12 @@ main (int argc, char *argv[])
...
+	g_signal_connect (widget, "scroll-event",
+					  G_CALLBACK (gpm_dialog_page_scroll_event_cb),
```
Comment (raveit65) [inline src/gpm-statistics.c]: It gets dramatic , indents!
I do not really get why you have so heavy problems with using tabs and spaces.
And why you don't check it in your own repo in browser before pushing here.
https://github.com/tamplan/mate-power-manager/commit/7a731e4fd30210a62543f146d1757eb6924d4462

Believe me, this review isn't funny for me. I don't really like to post always the same simple issues.

Comment (tamplan) [reply src/gpm-statistics.c]: Capture du 2019-07-10 19-43-12
Sure it's not funny for me too, see the file i have, i don't want to push more modification than necessary.

**Comment by tamplan** [comment]
i have seen this from the link you give me:
`@@ -776,8 +783,8 @@ gpm_prefs_init (GpmPrefs *prefs)
 			return;
 		}
 
-		res = g_dbus_proxy_call_sync (proxy, "Get", 
-					      g_variant_new( "(ss)", 
+		res = g_dbus_proxy_call_sync (proxy, "Get",
+					      g_variant_new( "(ss)",
 							     "org.freedesktop.UPower",
 							     "LidIsPresent"),
 					      G_DBUS_CALL_FLAGS_NONE,
`
This is not from me, can i close this PR and purpose it later once the file clean (tabs replaced by spaces)?

**Comment by raveit65** [comment]
Sorry, wrong link, but this one is from your repo
https://github.com/tamplan/mate-power-manager/commit/5f862d8f4c75bd66a1a79f72758b2cf68530d272.diff
Please do nothing, i will fix your commit before merging.

**Comment by raveit65** [comment]
merged https://github.com/mate-desktop/mate-power-manager/commit/67b0883a92607d52ece6ab7e8f8869e7f55bab29

**Pull request closed without merging**